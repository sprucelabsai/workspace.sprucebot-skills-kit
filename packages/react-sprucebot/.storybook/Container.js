import React, { Component } from 'react';

export default class Container extends Component {
  render() {
    const { story } = this.props;

    return (
      <React.StrictMode>
        <div
          data-floating-menu-container
          role="main"
          className="wf-active storybook-container" // NOTE: This tricks icons into appearing
          style={{
            padding: '3em',
            display: 'flex',
            flexDirection: 'column',
            alignItems: 'center',
          }}>
          {story()}
        </div>
        <input
          aria-label="input-text-offleft"
          type="text"
          className="bx--visually-hidden"
        />
      </React.StrictMode>
    );
  }
}
