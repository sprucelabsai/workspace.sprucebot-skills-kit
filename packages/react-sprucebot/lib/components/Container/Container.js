"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var Container = _react.default.forwardRef(function (_ref, ref) {
  var className = _ref.className,
      props = (0, _objectWithoutProperties2.default)(_ref, ["className"]);
  return _react.default.createElement("div", (0, _extends2.default)({}, props, {
    ref: ref,
    className: "container ".concat(className || '')
  }));
});

var _default = Container;
exports.default = _default;