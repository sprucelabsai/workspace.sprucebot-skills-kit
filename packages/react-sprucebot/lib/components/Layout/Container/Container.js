"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _classnames = _interopRequireDefault(require("classnames"));

var Container = function Container(props) {
  var children = props.children,
      className = props.className,
      size = props.size,
      center = props.center;
  var parentClass = (0, _classnames.default)('l-container', className, {
    'l-container-small': size === 'small',
    'l-container-medium': size === 'medium'
  });
  return _react.default.createElement("div", {
    className: parentClass
  }, props.children);
};

Container.defaultProps = {
  center: false,
  className: ''
};
var _default = Container;
exports.default = _default;