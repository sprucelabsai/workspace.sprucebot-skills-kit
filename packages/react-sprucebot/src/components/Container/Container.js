import React from 'react'

const Container = () => {
	return <div className="container" />
}

export default Container
