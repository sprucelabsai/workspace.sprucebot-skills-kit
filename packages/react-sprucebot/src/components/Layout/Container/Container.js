// @flow
import React from 'react'
import cx from 'classnames'

type Props = {
	children: React.Node,
	className: ?string,
	size: 'small' | 'medium',
	center: ?boolean
}

const Container = (props: Props) => {
	const { children, className, size, center } = props
	const parentClass = cx('l-container', className, {
		'l-container-small': size === 'small',
		'l-container-medium': size === 'medium'
	})
	return <div className={parentClass}>{props.children}</div>
}

Container.defaultProps = {
	center: false,
	className: ''
}

export default Container
