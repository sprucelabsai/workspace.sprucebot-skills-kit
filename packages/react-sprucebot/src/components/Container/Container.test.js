import React from 'react'
import Container from './Container'
import { shallow } from 'enzyme'

describe('Container tests', () => {
	it('Should match the snapshot', () => {
		const renderedComponent = shallow(<Container />)

		expect(renderedComponent).toMatchSnapshot()
	})
})
