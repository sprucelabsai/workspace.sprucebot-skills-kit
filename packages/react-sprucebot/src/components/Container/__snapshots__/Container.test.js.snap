// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container Component it renders 1`] = `
<div
  className="Container-md3jvi-0 container cqhPkv"
>
  Container children
</div>
`;
